subsys/fs/nvs/nvs.c

```diff
@@ -854,6 +863,11 @@ static int nvs_startup(struct nvs_fs *fs)
 		fs->ate_wra &= ADDR_SECT_MASK;
 		fs->ate_wra += (fs->sector_size - 2 * ate_size);
 		fs->data_wra = (fs->ate_wra & ADDR_SECT_MASK);
+#ifdef CONFIG_NVS_LOOKUP_CACHE
+		for (int i = 0; i < CONFIG_NVS_LOOKUP_CACHE_SIZE; i++) {
```

Comment:
This is a tricky bit of code, at first sight I don't think it will break but it should be thoroughly checked. It definitively needs some comment e.g. "temporarily set the lookup cache to the end of the fs, the cache will be rebuild".